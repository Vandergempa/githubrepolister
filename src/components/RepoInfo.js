import React, { Fragment } from "react"
import AccordionCard from './AccordionCard'

import Navbar from 'react-bootstrap/Navbar'
import Breadcrumb from 'react-bootstrap/Breadcrumb'
import Accordion from 'react-bootstrap/Accordion'

const RepoInfo = (props) => {
  const repoDetails = props.location.state.details
  return (
    <Fragment>
      <header>
        <Navbar bg="light" expand="sm">
          <img className="github-icon mr-2" src="https://unicons.iconscout.com/release/v2.0.1/svg/line/github.svg" alt="Kiwi standing on oval" />
          <Navbar.Brand href="/">GitHub Repo Lister</Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse className="justify-content-end">
            <Navbar.Text className="cursor-pointer" onClick={() => window.history.back()}>
              Back
            </Navbar.Text>
          </Navbar.Collapse>
        </Navbar>
        <Breadcrumb bg="light">
          <Breadcrumb.Item active>Main</Breadcrumb.Item>
          <Breadcrumb.Item active>{repoDetails.owner.login}</Breadcrumb.Item>
          <Breadcrumb.Item active>{repoDetails.name}</Breadcrumb.Item>
        </Breadcrumb>
      </header>
      <section className="w-100">
        <h1 className="text-center mb-5 mt-5 display-3 custfontfamily">{repoDetails.name}</h1>
        <Accordion className="mb-2 cursor-pointer col-6 mr-auto ml-auto" defaultActiveKey="0">
          <AccordionCard eventKey="0" title="Description" body={repoDetails.description} />
          <AccordionCard eventKey="1" title="Repo URL" body={<a href={repoDetails.html_url}>{repoDetails.html_url}</a>} />
          <AccordionCard eventKey="2" title="Homepage" body={<a href={repoDetails.homepage}>{repoDetails.homepage}</a>} />
          <AccordionCard eventKey="3" title="Created At" body={repoDetails.created_at} />
          <AccordionCard eventKey="4" title="Updated At" body={repoDetails.updated_at} />
          <AccordionCard eventKey="5" title="Clone URL - SSH" body={repoDetails.ssh_url} />
          <AccordionCard eventKey="6" title="Main language" body={repoDetails.language} />
          <AccordionCard eventKey="7" title="Fork count" body={repoDetails.forks} />
          <AccordionCard eventKey="8" title="Watchers count" body={repoDetails.watchers} />
        </Accordion>
      </section>
    </Fragment>
  )
}

export default RepoInfo