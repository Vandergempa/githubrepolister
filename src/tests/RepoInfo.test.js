import React from 'react';
import { shallow } from 'enzyme';
import RepoInfo from '../components/RepoInfo';

let wrapper

beforeEach(() => {
  const props = {
    location: {
      state: {
        details: {
          created_at: "1992",
          description: "Something",
          forks: 0,
          html_url: "www.github.com",
          homepage: "www.something.com",
          language: "HTML",
          name: "Tom",
          owner: {
            login: "Gempa"
          },
          ssh_url: "something",
          updated_at: "1995",
          watchers: 0
        }
      }
    }
  }
  wrapper = shallow(
    <RepoInfo
      {...props}
    />
  );
});

test('Should render RepoInfo component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});