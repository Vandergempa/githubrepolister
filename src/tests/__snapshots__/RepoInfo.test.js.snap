// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render RepoInfo component correctly 1`] = `
<Fragment>
  <header>
    <Navbar
      bg="light"
      collapseOnSelect={false}
      expand="sm"
      variant="light"
    >
      <img
        alt="Kiwi standing on oval"
        className="github-icon mr-2"
        src="https://unicons.iconscout.com/release/v2.0.1/svg/line/github.svg"
      />
      <NavbarBrand
        href="/"
      >
        GitHub Repo Lister
      </NavbarBrand>
      <NavbarToggle
        aria-controls="basic-navbar-nav"
        label="Toggle navigation"
      />
      <NavbarCollapse
        className="justify-content-end"
      >
        <NavbarText
          className="cursor-pointer"
          onClick={[Function]}
        >
          Back
        </NavbarText>
      </NavbarCollapse>
    </Navbar>
    <Breadcrumb
      bg="light"
      label="breadcrumb"
      listProps={Object {}}
    >
      <BreadcrumbItem
        active={true}
      >
        Main
      </BreadcrumbItem>
      <BreadcrumbItem
        active={true}
      >
        Gempa
      </BreadcrumbItem>
      <BreadcrumbItem
        active={true}
      >
        Tom
      </BreadcrumbItem>
    </Breadcrumb>
  </header>
  <section
    className="w-100"
  >
    <h1
      className="text-center mb-5 mt-5 display-3 custfontfamily"
    >
      Tom
    </h1>
    <ForwardRef
      className="mb-2 cursor-pointer col-6 mr-auto ml-auto"
      defaultActiveKey="0"
    >
      <AccordionCard
        body="Something"
        eventKey="0"
        title="Description"
      />
      <AccordionCard
        body={
          <a
            href="www.github.com"
          >
            www.github.com
          </a>
        }
        eventKey="1"
        title="Repo URL"
      />
      <AccordionCard
        body={
          <a
            href="www.something.com"
          >
            www.something.com
          </a>
        }
        eventKey="2"
        title="Homepage"
      />
      <AccordionCard
        body="1992"
        eventKey="3"
        title="Created At"
      />
      <AccordionCard
        body="1995"
        eventKey="4"
        title="Updated At"
      />
      <AccordionCard
        body="something"
        eventKey="5"
        title="Clone URL - SSH"
      />
      <AccordionCard
        body="HTML"
        eventKey="6"
        title="Main language"
      />
      <AccordionCard
        body={0}
        eventKey="7"
        title="Fork count"
      />
      <AccordionCard
        body={0}
        eventKey="8"
        title="Watchers count"
      />
    </ForwardRef>
  </section>
</Fragment>
`;
